### PR TITLE
Add calico configurations

### PR DESCRIPTION
**Pull request by mainred**
Add calico configurations into terraform and make several small fixes. 

fix: [DCOS-62550](https://jira.mesosphere.com/browse/DCOS-62550) integration calico configuration into terraform

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -326,3 +326,38 @@ variable "public_agents_acm_cert_arn" {
   description = "ACM certifacte to be used for the public agents load balancer"
   default     = ""
 }
+
+variable "dcos_calico_network_cidr" {
```
Comment (fatz) [inline variables.tf]: they are already defined in dcos_core_variables.tf defining them again leads to duplicate variable errors

**Review by fatz — CHANGES_REQUESTED** [review]
please define variables only once. dcos_core_variables.tf seems to be the best place

**Comment by fatz** [comment]
the minimum install module version should be `version = "~> 0.2.4"` as previous releases do not support the new vars

File: variables.tf (hunk 1 above)
Comment (mainred) [reply variables.tf]: Done, thanks.

**Comment by mainred** [comment]
@fatz , the required version is updated, please take another look.

File: dcos_core_variables.tf [hunk 2]
```diff
@@ -523,3 +523,33 @@ variable "dcos_versions_service_url" {
...
+
+variable "dcos_calico_vxlan_vni" {
+  default     = ""
+  description = "The virtual network ID used for calico VXLAN. (optional)"
+}
+
+variable "dcos_calico_vxlan_mtu" {
+  default     = ""
+  description = "The MTU to set on the Calico VXLAN tunnel device. (optional)"
+}
```
Comment (fatz) [inline dcos_core_variables.tf:550]: the `dcos_calico_vxlan_port` is missing here now so we get an 

```
Error: module 'dcos-install': unknown variable referenced: 'dcos_calico_vxlan_port'; define it with a 'variable' block
```

**Review by fatz — CHANGES_REQUESTED** [review]
now one variable is missing please have a look

File: dcos_core_variables.tf (hunk 2 above)
Comment (mainred) [reply dcos_core_variables.tf:550]: Done, thanks.

File: dcos_core_variables.tf [hunk 3]
```diff
@@ -523,3 +523,38 @@ variable "dcos_versions_service_url" {
   default     = "https://versions.d2iq.com"
   description = "DC/OS Versions Service allows to identify DC/OS versions"
 }
+
+variable "dcos_calico_network_cidr" {
+  default     = ""
```
Comment (jkoelker) [inline dcos_core_variables.tf:528]: What do we think about defaulting to something like `172.29.0.0/16`? This is a fairly unused subnet, being at the high end of `172.16.0.0/12` but not at the very end of it. I think having a good default while keeping the ability to override here would make the end user experience better.

Comment (mainred) [reply dcos_core_variables.tf:528]: I second you on making the end-user experience better if possible, but I insist on using mandatory calico subnet configuration, the side effect of using the overlapped networks we had mentioned in https://github.com/dcos/dcos/pull/6244#discussion_r340788009
but we need to draw a balance between making 95+% of our customers happy, and the left ones in anger.
I am happy to discuss this.

**Pull request merged**